pdx-extract/app/main.py

```diff
@@ -0,0 +1,175 @@
+import time
+
+import requests
+
+from app import config
+from app import pyppeteer
+from app import sanitize
+from app import storage
+from app import urls
+from app.logger import logger
+
+
+def authenticate_current_session(term, unique_session_id, cookies):
+    """Make a POST request that will authenticate the user with this JSESSIONID
+    and uniqueSessionId and enable the sched_page GET request to return JSON
+
+    :param term: Term dictionary with code and description keys.
+    :param unique_session_id: Unique session id generated by the page which
+                              allows authentication.
+    :param cookies: Cookies of the previous requests.
+    """
+
+    payload = {
+        "dataType": "json",
+        "endDatepicker": "",
+        "startDatepicker": "",
+        "studyPath": "",
+        "studyPathText": "",
+        "term": term["code"],
+        "uniqueSessionId": unique_session_id,
+    }
+    return requests.post(
+        urls.SEARCH_URL,
+        headers={"referer": urls.INIT_URL},
```

Comment:
Gnit: as a best practice, use canonical HTTP header capitalization. I know, the HTTP standards say it's case insensitive, and it rarely makes a difference, but at those times when it does make a difference, boy will it save you a lot of debugging time. Speaking from horrendous experience.

`referer` -> `Referer`

Reply:
Something else to note -- this kind of automation is strictly necessary to bypass CSRF protections, which can get people mad at you if anyone ever uses your script for evil. Pretty safe in this case, but always be considerate of the terms of service, in particular under what circumstances automated systems are allowed to access the platform you're developing against.

Reply:
Definitely a pro tip here! Thanks for the heads up.

I know that PSU data is public, or it should be. But still, I understand I'm bypassing their intent. Don't use this for evil!